lib/blockchain/chaindb.js

```diff
@@ -225,6 +227,9 @@ class ChainDB {
     this.current = this.db.batch();
     this.pending = this.state.clone();
 
+    if (this.blocks)
```

Comment:
Do we have a test plan for cases where `chain` and `blocks` directories are out-of-sync due to factors outside of the process? For e.g. either directory gets corrupted and has to be `rm`ed and restarted from scratch. This wasn't an issue before as there was only one db.

Reply:
Well, you could say it was a case previously as well, because ldb has many files and some of them could be removed by outside process.

Currently, there's no such thing - but I think in the future we can do better and this comment by @pinheadmz goes well with this issue: https://github.com/handshake-org/hsd/pull/612#discussion_r660906283
Making blockstore files more flexible.

Reply:
Yep sounds good. Something to note for later.